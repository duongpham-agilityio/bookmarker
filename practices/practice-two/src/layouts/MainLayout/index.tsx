import { Suspense } from 'react';
import { SWRConfig } from 'swr';
import { Outlet } from 'react-router-dom';

// Layout
import Header from 'layouts/Header';

// Context
import { FormProvider, PopupProvider, ToastProvider } from 'contexts';

// Helpers
import { fetcher } from 'helpers';

const MainLayout = () => {
  return (
    <SWRConfig
      value={{
        fetcher,
      }}
    >
      <ToastProvider>
        <PopupProvider>
          <FormProvider>
            <Header />
            <Suspense>
              <Outlet />
            </Suspense>
          </FormProvider>
        </PopupProvider>
      </ToastProvider>
    </SWRConfig>
  );
};

export default MainLayout;
