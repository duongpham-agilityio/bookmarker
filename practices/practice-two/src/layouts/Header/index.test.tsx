import { render } from '@testing-library/react';

import { Header } from 'layouts';

jest.mock('react-router-dom', () => ({
  Link: () => <a href="">header</a>,
}));

describe('Header', () => {
  it('match snapshot', () => {
    const { asFragment } = render(<Header />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('render', () => {
    const { getByText } = render(<Header />);

    expect(getByText(/header/i)).toBeInTheDocument();
  });
});
