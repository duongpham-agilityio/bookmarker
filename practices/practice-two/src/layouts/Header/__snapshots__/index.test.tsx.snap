// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header match snapshot 1`] = `
<DocumentFragment>
  <header>
    <a
      href=""
    >
      header
    </a>
    <h2
      class="undefined undefined "
    >
      Hello! Welcome to the Book Marker
    </h2>
  </header>
</DocumentFragment>
`;
